src/picklescan/scanner.py

```diff
@@ -307,12 +327,35 @@ def scan_pickle_bytes(data: IO[bytes], file_id, multiple_pickles=True) -> ScanRe
     return _build_scan_result_from_raw_globals(raw_globals, file_id)
 
 
+# XXX: it appears there is not way to get the byte stream for a given file within the 7z archive and thus forcing us to unzip to disk before scanning
+def scan_7z_bytes(data: IO[bytes], file_id) -> ScanResult:
+    result = ScanResult([])
+
+    archive = py7zr.SevenZipFile(data, mode="r")
+    file_names = archive.getnames()
+    _log.debug("Files in 7z archive %s: %s", file_id, file_names)
+    rand_id = "".join(random.choice(string.ascii_letters) for _ in range(8))
+    tmp_path = f"/tmp/archive-{rand_id}"
+    archive.extractall(tmp_path)
+    for file_name in file_names:
+        file_path = f"{tmp_path}/{file_name}"
+        file_ext = os.path.splitext(file_name)[1]
+        if file_ext in _pickle_file_extensions:
```

Comment:
Instead of extracting all the files first and then filtering out files based on extensions, it would be more efficient (CPU + disk space) to filter out file names based on extensions first and then only extract those that match. Related code snippet from the py7zr docs (slightly simplified):
```python
filter_pattern = re.compile(r'scripts.*')
with SevenZipFile('archive.7z', 'r') as zip:
     allfiles = zip.getnames()
     targets = [f if filter_pattern.match(f) for f in allfiles]
     zip.extract(targets=targets, recursive=True)
```
https://py7zr.readthedocs.io/en/latest/api.html#py7zr.SevenZipFile.extract